frontend/src/components/pages/TranslationPage.tsx

```diff
@@ -1,33 +1,193 @@
 import React, { useState } from "react";
+import { gql, useMutation, useQuery } from "@apollo/client";
+
 import "./TranslationPage.css";
 import { useParams } from "react-router-dom";
+import Cell from "../translation/Cell";
+import EditableCell from "../translation/EditableCell";
 import TranslationProgressBar from "../translation/TranslationProgressBar";
+import CheckmarkIcon from "../../assets/checkmark.svg";
+import CommentIcon from "../../assets/comment_no_number.svg";
 
 type TranslationPageProps = {
-  storyId: string | undefined;
-  storyTranslationId: string | undefined;
+  storyIdParam: string | undefined;
+  storyTranslationIdParam: string | undefined;
+};
+
+type StoryLine = {
+  lineIndex: number;
+  originalContent: string;
+  translatedContent?: string;
+  storyTranslationContentId?: number;
+};
+
+type Content = {
+  id: number;
+  lineIndex: number;
+  content: string;
+};
+const GET_STORY_CONTENTS = (storyId: number, storyTranslationId: number) => gql`
+  query {
+    storyById(id: ${storyId}) {
+      contents {
+        id
+        lineIndex
+        content
+      }
+    }
+    storyTranslationById(id: ${storyTranslationId}) {
+      translationContents {
+        id
+        lineIndex
+        content: translationContent
+      }
+    }
+  }
+`;
+
+type UpdateTranslationResponse = {
+  story: { lineIndex: number };
 };
+const UPDATE_TRANSLATION = gql`
+  mutation updateStoryTranslationContentById(
+    $storyTranslationContentId: Int!
+    $newTranslationContent: String!
+  ) {
+    updateStoryTranslationContentById(
+      storyTranslationContentData: {
+        id: $storyTranslationContentId
+        translationContent: $newTranslationContent
+      }
+    ) {
+      story {
+        lineIndex
+      }
+    }
+  }
+`;
 
 const TranslationPage = () => {
-  const { storyId, storyTranslationId } = useParams<TranslationPageProps>();
-  // TODO: set percentageComplete based on return from get translation query,
-  // and update when edits made to page
-  // const [percentageComplete, setPercentageComplete] = useState(0);
```

Comment:
In the current state, the translation progress bar is static at 25%. Do you want to integrate the state or save this for a separate ticket?

Reply:
If it makes it easier, I can work on implementing the rest of the progress bar logic once this PR is merged; will need to figure out how to update the progress on updates as well in the frontend